src/iter/mod.rs

```diff
@@ -1671,6 +1716,33 @@ pub trait IndexedParallelIterator: ParallelIterator {
 ///
 /// [`ParallelIterator`]: trait.ParallelIterator.html
 /// [`collect()`]: trait.ParallelIterator.html#method.collect
+///
+/// # Examples
```

Comment:
fwiw I often think that the idiomatic style for iterators out to be `from_iter` (and hence `from_par_iter`). I like that writing 

```rust
let x =
    BlackHole::from_par_iter(...);
```

kind of "automatically" gives the needed type-annotation. But oh well.